components.go

```diff
@@ -603,6 +603,9 @@ func ParseComponent(cs *CalendarStream, startLine *BaseProperty) (ComponentBase,
 		}
 		line := ParseProperty(*l)
 		if line == nil {
+			if strings.HasPrefix(string(*l), "X-") {
+				continue
```

Comment:
as i mention before, your code fails is case of custom ics fields(https://icalendar.org/iCalendar-RFC-5545/3-8-8-2-non-standard-properties.html). However it`s not necessarily, so i think we can skip this properties in such way.